### ISSUE TITLE
Add heroes functionality won't compile in Tour of Heroes tutorial part 7 (HTTP) 

### ISSUE THREAD

**Issue by gbgames** [angular/angular#20177]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ x] Bug report  
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior

Following the tutorial's instructions, once I add the create() method in HeroService, I see:

Failed to compile.

And among the errors: 

hero.service.ts (47,6): Type 'Promise<Response>' is not assignable to type 'Promise<Hero>'.
  Type 'Response' is not assignable to type 'Hero'.
    Property 'id' is missing in type 'Response'.

## Expected behavior

I expected that the tutorial would be complete and allow me to do what it says: "Refresh the browser and create some heroes."

## Minimal reproduction of the problem with instructions

I followed the tutorial for HTTP. 

## What is the motivation / use case for changing the behavior?

Currently I am not sure what I am supposed to do. The live example seems to run just fine, but the code doesn't look much different (although I note that the json().data line still exists, while I found that I needed to remove the data part in my own code). 

As I'm new to Angular, I'm not sure if this issue is due to the environment or an omission in the tutorial.

## Environment

<pre><code>
Angular version: 4.4.6

Browser:
- [ ] Chrome (desktop) version XX
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [x] Firefox version 56.0.1
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: 6.11.5  
- Platform:  Mac 

Others:

</code></pre>

**Comment by SanderElias** [comment]
@gbgames Do you have a link to the tutorial you were following? (yes, there is more as one, and there are different versions)

**Comment by gbgames** [comment]
https://angular.io/tutorial/toh-pt6

**Comment by kapunahelewong** [comment]
@gbgames If you'd put your version of the app on GitHub, I'd be happy to look at it and see if I can find the issue.

**Comment by praveeno** [comment]
let me know if i can help

**Comment by gbgames** [comment]
Thank you all for your offers for help. While I did run into a number of discrepancies trying to follow the tutorial, this time the error was mine. I've been busy and haven't been able to look into this issue sooner. 

Now that I know Angular a bit more, I took a look in the hopes that I could figure out the problem, and I noticed a stray ] in my code. 

I deleted it, and the code compiles. My forehead hurts from smacking it. 

Thanks again.
